docs/presentation/hierarchies/FilteringLimiting.md

```diff
@@ -0,0 +1,129 @@
+
+### Workflow
+
+1. Hierarchy levels that support filtering have a filter icon at the right of the parent node when it's hovered:
+
+   ![Filter icon on filterable parent node](./media/filtering-limiting-workflow-1.png)
+
+2. When the button is pressed, the filtering dialog shows up:
+
+   ![Filtering dialog](./media/filtering-limiting-workflow-2.png)
+
+   The dialog allows filtering the results by class and properties of elements w hich the hierarchy level consists of and doesn't display properties of classes that don't exist in the hierarchy level. In addition, selecting a class filters the available property list to show only properties for that class of elements. And it works in the opposite direction too - selecting a property filters the classes to show only the ones that have the property.
+
+3. Pressing the "Apply" button filters the child hierarchy level:
+
+   ![Applied filter](./media/filtering-limiting-workflow-3.png)
+
+   Note that after applying the filter the parent node has the highlighted "filter" and "clear filter" buttons displayed at the right even when the node is not hovered.
+
+4. In case the filter results in 0 nodes, the "no matches" message is displayed:
+
+   ![No matches for current filter](./media/filtering-limiting-workflow-4.png)
+
+   The "clear filter" button can be pressed to return to initial, unfiltered state. The "filter" button can be pressed to edit the applied filter.
+
+### Limitations
+
+There's a limited set of presentation rules, their attributes or attribute values that don't allow us to efficiently apply the filter:
+
+- `hideExpression` attribute on the hierarchy specifications [Instance nodes of specific classes](./InstanceNodesOfSpecificClasses.md#attribute-hideexpression), [Related instance nodes](./RelatedInstanceNodes.md#attribute-hideexpression) and [Custom query instance nodes](./CustomQueryInstanceNodes.md#attribute-hideexpression).
+
+- Usage of the `parent` ECExpression symbol in `instanceFilter` attribute of either [Instance nodes of specific classes](./InstanceNodesOfSpecificClasses.md#attribute-instancefilter) or [Related instance nodes](./RelatedInstanceNodes.md#attribute-instancefilter) specifications.
+
+- Usage of some deprecated, undocumented rules or attributes.
+
+When any of the above is used to create a hierarchy level, filtering for that level is going to be disabled.
+
+### Setting up hierarchy level filtering
+
+Setting up hierarchy level filtering is as simple as supplying a presentation-specific tree renderer to the [ControlledTree]($components-react) component:
+
+```tsx
+function MyTree(props: MyTreeProps) {
+  const { nodeLoader, onItemsRendered } = usePresentationTreeNodeLoader({
+    // supply nodes loader props
+  });
+  const eventHandler = useUnifiedSelectionTreeEventHandler({ nodeLoader });
+  const treeModel = useTreeModel(nodeLoader.modelSource);
+
+  // create presentation-specific tree renderer
+  const treeRenderer = React.useCallback(
```

Comment:
I don't think there is a need to have a stable reference of `treeRenderer` so `useCallback` can be removed to simplify this example.

Reply:
simplified